src/platform/common/process/environmentActivationService.node.ts

```diff
@@ -110,33 +110,20 @@ export class EnvironmentActivationService implements IEnvironmentActivationServi
             | 'failedToGetActivatedEnvVariablesFromPython'
             | 'failedToGetCustomEnvVariables' = 'emptyVariables';
         let failureEx: Error | undefined;
-        let [env, customEnvVars] = await Promise.all([
-            this.apiProvider.getApi().then((api) =>
-                api
-                    .getActivatedEnvironmentVariables(resource, serializePythonEnvironment(interpreter)!, false)
-                    .catch((ex) => {
-                        traceError(
-                            `Failed to get activated env variables from Python Extension for ${getDisplayPath(
-                                interpreter.uri
-                            )}`,
-                            ex
-                        );
-                        reasonForFailure = 'failedToGetActivatedEnvVariablesFromPython';
-                        return undefined;
-                    })
-            ),
-            this.envVarsService.getCustomEnvironmentVariables(resource, 'RunPythonCode').catch((ex) => {
```

Comment:
With the new Python API we do not need to merge any of the custom env variables.